great_expectations/data_context/data_context.py

```diff
@@ -3877,13 +3877,29 @@ def _test_instantiation_of_misc_class_from_yaml_config(
         data_connector_anonymizer: DataConnectorAnonymizer = DataConnectorAnonymizer(
             self.data_context_id
         )
-        if store_anonymizer.get_parent_class(store_obj=instantiated_class) is not None:
+
+        store_parent_class: Optional[str] = store_anonymizer.get_parent_class(
+            store_obj=instantiated_class
+        )
+        datasource_parent_class: Optional[str] = datasource_anonymizer.get_parent_class(
+            config=config
+        )
+        checkpoint_parent_class: Optional[str] = checkpoint_anonymizer.get_parent_class(
+            config=config
+        )
+        data_connector_parent_class: Optional[
+            str
+        ] = data_connector_anonymizer.get_parent_class(config=config)
+
+        if store_parent_class is not None and store_parent_class.endswith("Store"):
             store_name: str = name or config.get("name") or "my_temp_store"
             store_name = instantiated_class.store_name or store_name
             usage_stats_event_payload = store_anonymizer.anonymize_store_info(
                 store_name=store_name, store_obj=instantiated_class
             )
-        elif datasource_anonymizer.get_parent_class(config=config) is not None:
+        elif datasource_parent_class is not None and datasource_parent_class.endswith(
+            "Datasource"
+        ):
```

Comment:
Because `get_parent_class` is now generic (due to the removal of `self._ge_classes`), we need to check that the return value has the appropriate suffix before going down a particular branch.